internal/captain/template.go

```diff
@@ -0,0 +1,93 @@
+package captain
```

Comment:
This file is kind of all over the place. Can you please organize your code a little. I'll try to make notes to call out particular things but that's just to hopefully give some idea on what can be improved. Really this just needs some architecture put into it. At the moment this looks like prototype code.

Reply:
Agreed. What's left in this file _should_ just be related to templates